sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -523,6 +523,23 @@ int64_t CurlSession::Read(Azure::Core::Context const& context, uint8_t* buffer,
   totalRead = ReadFromSocket(buffer, static_cast<size_t>(readRequestLength));
   this->m_sessionTotalRead += totalRead;
 
+  // Reading 0 bytes means closed connection.
+  // For known content length and chunked response, this means there is nothing else to read from
+  // server or lost connection before getting full response.
+  // For unknown response size, it means the end of response and it's fine.
+  if (totalRead == 0 && (this->m_contentLength > 0 || this->m_isChunkedResponseType))
+  {
+    auto expectedToRead = this->m_isChunkedResponseType ? this->m_chunkSize : this->m_contentLength;
+    if (this->m_sessionTotalRead < expectedToRead)
+    {
+      throw Azure::Core::Http::TransportException(
+          "Connection closed before getting full response or response is less than expected. "
+          "Expected response length = "
+          + std::to_string(expectedToRead)
```

Comment:
Though such detail is useful, is there any information disclosure concerns with showing this as part of the exception message?

Reply:
no. And most of the cases this won't be bubble up to customers.
Most of the cases this exception will be catch by retry policy or a reliableSstreamReader (when we handle stream to customer).
If the issue was a quick network disconnect, everything will work fine after retry.
If there is a total network lost, the next error bubble up to customer will be COULD NOT RESOLVE HOST, or something like that, but not this one.

The one weird case when this can bubble to customer is if an sporadic and quick connection lost occurs on every retry!